src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
 }
 
 export class Triangle {
+  public shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.validateTriangle();
+  }
+
+  validateTriangle(): void {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Figure lenght must be > 0');
+    }
+
+    const sidesArray: number[] = [this.a, this.b, this.c];
```

Comment:
don't add the data type to the name of the variable unless you really need to
```suggestion
    const sides: number[] = [this.a, this.b, this.c];
```

Reply:
Done